src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
Alright, I'm fairly certain that the above will work pretty well for _across_ moves.
Using `mem::swap` we could swap the following
- parent
- children
- data

After that, we need to remove those `NodeId`s from their original parents' children (and add them to their new parents' children), and then swapping the position of those two `Node`s in `self.nodes` should give us what we want. (I drew this out on paper and it seemed to make sense, but I haven't tried it in code yet).

What I'm not sure about is how to use similar logic to deal with up/down swaps.
In fact, now that I'm trying to think it through, I'm not even sure what we want it to do in the case where there's one or more `Node`s between the two being swapped (vertically).

Starting with:
```
   A
  / \
 B   C
    / \
   D   E
        \
         F
```
`swap(C,F)` could result in:
```
OPTION 1     OR     OPTION 2
   A                   A
  / \                 / \
 B   F               B   F
      \                   \
       E                   C
        \                 / \
         C               D   E
        /
       D
```
In `OPTION 1` above, we would keep the "swap parents" step but modify the "swap children" step (to leave `E` where it is instead of it traveling down with `C`).
In `OPTION 2` above, we would keep the "swap children" step but modify the "swap parents" step (to take 'E' down with 'C' which means that `E` then becomes the child of `F` instead of `E`).

As you can see this is very similar to the `move_node_to_parent` discussion.  Since we went with behavior similar to `OPTION 1` on that method, I'm thinking it makes sense to do the same here.  Do you have any thoughts on this @Drakulix ?